R/extract_rhs.R

```diff
@@ -188,3 +188,55 @@ extract_subscripts <- function(primary, full_term_v) {
   )
   out
 }
+
+#' Generic function for wrapping the RHS of a model equation in something, like
+#' how the RHS of probit is wrapped in φ()
+#'
+#' @keywords internal
+#'
+#' @param model A fitted model
+#' @param tex The TeX version of the RHS of the model (as character), built as
+#'   \code{rhs_combined} or \code{eq_raw$rhs} in \code{extract_eq()}
+#' @param \dots additional arguments passed to the specific extractor
+
+wrap_rhs <- function(model, tex, ...) {
+  UseMethod("wrap_rhs", model)
+}
+
+#' @keywords internal
+wrap_rhs.default <- function(model, tex, ...) {
+  return(tex)
+}
+
+#' @keywords internal
+wrap_rhs.glm <- function(model, tex, ...) {
+  if (model$family$link == "probit") {
+    rhs <- paste0("\\phi(", tex, ")")
+  } else {
+    rhs <- tex
+  }
+
+  return(rhs)
+}
+
+#' @keywords internal
+wrap_rhs.polr <- function(model, tex, ...) {
+  if (model$method == "probit") {
+    rhs <- paste0("\\phi(", tex, ")")
+  } else {
+    rhs <- tex
+  }
+
+  return(rhs)
+}
+
+#' @keywords internal
+wrap_rhs.clm <- function(model, tex, ...) {
+  if (model$info$link == "probit") {
+    rhs <- paste0("\\phi(", tex, ")")
+  } else {
+    rhs <- tex
+  }
+
+  return(rhs)
```

Comment:
Okay so these ones we could probably do something like this, right?

```r

#' @keywords internal
wrap_rhs.glm <- function(model, tex, ...) {
  if (model$family$link == "probit") {
    rhs <- paste0("\\phi(", tex, ")")
  } else {
    rhs <- tex
  }

  return(rhs)
}

#' @keywords internal
wrap_rhs.polr <- function(model, tex, ...) {
  wrap_rhs.glm(model, tex, ...)
}

#' @keywords internal
wrap_rhs.clm <- function(model, tex, ...) {
  wrap_rhs.glm(model, tex, ...)
}
```

That maybe makes them a little less general but seems easy to modify in the future if we want. Maybe I'm not thinking about this right though and there's reasons we wouldn't want to do that? I just prefer not to repeat code if I can help it.

Reply:
Yeah, I don't like the repetition, but the three model types store their link information in different places `model$family$link` vs. `model$method` vs. `model$info$link`, so we'd still need specific S3 classes to deal with each different model type. We could maybe abstract it somewhat and make a new S3 class called `get_link` or something, and then if `get_link == "probit"`, do the wrapping?

Reply:
Oh right... I missed that. Yeah, maybe a `get_link` function would be helpful. All of these inconsistencies reminds me of the discussion around the **parsnip** package and why it is needed/helpful.